Jacky's Pi Sys/pi-systems_sensor-reader.py

```diff
@@ -15,42 +15,45 @@ class SensorSubsystem(subsys.Subsystem):
     class Procedure(Enum):
         GetSensorData = 1
 
-    def __init__(self, name=None, thread_id=None, addresses=None):
+    def __init__(self, name=None, thread_id=None, addresses=None, segment_addresses=None):
         super().__init__(name=name, thread_id=thread_id, loop_delay_ms=2000)
 
+        self.segment_addresses = segment_addresses
         # namedtuple is temporarily a dict for pickling purposes.
         self.sensor_data = {}  # self.SensorData(0,0,0,0,0)
         # sets addresses to a list of 1 or more addresses
         self.addresses = addresses if isinstance(addresses,
                                                  list) else [addresses]
         self.print_updates = False
+        # initialize readings lists in constructor
+        # index 0 is oldest reading
+        self.O2_readings = []
+        self.C02_readings = []
+        self.temp_readings = []
+        self.humidity_readings = []
+        self.pressure_readings = []
+
+        keep_track_seconds = 5
```

Comment:
keep_track_seconds never leaves the __init__ function scope, as it is local. You probably want to do something like _self.keep_track_seconds = 5_.